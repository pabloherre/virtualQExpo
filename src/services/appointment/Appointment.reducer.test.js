import appointmentReducer from './Appointment.reducer';
import { APPOINTMENT_ADD_SUCCESS, APPOINTMENT_FIND_FAILED, APPOINTMENT_FIND_PENDING, APPOINTMENT_FIND_SUCCESS } from './Appointment.actions';

describe('Appointment Reducer', () => {
  it('should should return initial state', async () => {
    const INITIAL_STATE = {
      data: null,
      error: null,
      loading: false
    };

    expect(appointmentReducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle APPOINTMENT_FIND_PENDING action', async () => {
    const INITIAL_STATE = {
      data: null,
      error: null,
      loading: true
    };

    expect(appointmentReducer(INITIAL_STATE, { type: APPOINTMENT_FIND_PENDING })).toMatchSnapshot();
  });

  it('should handle APPOINTMENT_FIND_SUCCESS action', async () => {
    const data = {
      name: 'test'
    };
    const INITIAL_STATE = {
      data,
      error: null,
      loading: true
    };

    expect(appointmentReducer(INITIAL_STATE, { type: APPOINTMENT_FIND_SUCCESS, payload: data })).toMatchSnapshot();
  });

  it('should handle APPOINTMENT_FIND_SUCCESS action', async () => {
    const INITIAL_STATE = {
      loading: false
    };

    expect(appointmentReducer(INITIAL_STATE, { type: APPOINTMENT_ADD_SUCCESS })).toMatchSnapshot();
  });

  it('should handle APPOINTMENT_FIND_FAILED action', async () => {
    const error = true;
    const INITIAL_STATE = {
      loading: false,
      data: null,
      error
    };

    expect(appointmentReducer(INITIAL_STATE, { type: APPOINTMENT_FIND_FAILED, payload: error })).toMatchSnapshot();
  });
});
