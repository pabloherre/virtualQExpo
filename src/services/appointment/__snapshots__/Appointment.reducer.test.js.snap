// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointment Reducer should handle APPOINTMENT_FIND_FAILED action 1`] = `
Object {
  "data": null,
  "error": true,
  "loading": false,
}
`;

exports[`Appointment Reducer should handle APPOINTMENT_FIND_PENDING action 1`] = `
Object {
  "data": null,
  "error": null,
  "loading": true,
}
`;

exports[`Appointment Reducer should handle APPOINTMENT_FIND_SUCCESS action 1`] = `
Object {
  "data": Object {
    "name": "test",
  },
  "error": null,
  "loading": false,
}
`;

exports[`Appointment Reducer should handle APPOINTMENT_FIND_SUCCESS action 2`] = `
Object {
  "loading": false,
}
`;

exports[`Appointment Reducer should should return initial state 1`] = `
Object {
  "data": null,
  "error": null,
  "loading": false,
}
`;
